File: web/src/api/sysUser.js
import service from '@/utils/request'
// @Summary 用户登录
// @Produce  application/json
// @Param data body {username:"string",password:"string"}
// @Router /base/login [post]
export const login = (data) => {
  return service({
    url: '/base/login',
    method: 'post',
    data: data
  })
}

// @Summary 获取验证码
// @Produce  application/json
// @Param data body {username:"string",password:"string"}
// @Router /base/captcha [post]
export const captcha = (data) => {
  return service({
    url: '/base/captcha',
    method: 'post',
    data: data
  })
}

// @Summary 用户注册
// @Produce  application/json
// @Param data body {username:"string",password:"string"}
// @Router /base/resige [post]
export const register = (data) => {
  return service({
    url: '/staff/admin_register',
    method: 'post',
    data: data
  })
}

// @Summary 修改密码
// @Produce  application/json
// @Param data body {username:"string",password:"string",newPassword:"string"}
// @Router /staff/changePassword [post]
export const changePassword = (data) => {
  return service({
    url: '/staff/changePassword',
    method: 'post',
    data: data
  })
}

// @Tags User
// @Summary 分页获取用户列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body modelInterface.PageInfo true "分页获取用户列表"
// @Success 200 {string} json "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /staff/getUserList [post]
export const getUserList = (data) => {
  return service({
    url: '/staff/getUserList',
    method: 'post',
    data: data
  })
}

// @Tags User
// @Summary 设置用户权限
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body api.SetUserAuth true "设置用户权限"
// @Success 200 {string} json "{"success":true,"data":{},"msg":"修改成功"}"
// @Router /staff/setUserAuthority [post]
export const setUserAuthority = (data) => {
  return service({
    url: '/staff/setUserAuthority',
    method: 'post',
    data: data
  })
}

// @Tags SysUser
// @Summary 删除用户
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.SetUserAuth true "删除用户"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"修改成功"}"
// @Router /staff/deleteUser [delete]
export const deleteUser = (data) => {
  return service({
    url: '/staff/deleteUser',
    method: 'delete',
    data: data
  })
}

// @Tags SysUser
// @Summary 设置用户信息
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.SysUser true "设置用户信息"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"修改成功"}"
// @Router /staff/setUserInfo [put]
export const setUserInfo = (data) => {
  return service({
    url: '/staff/setUserInfo',
    method: 'put',
    data: data
  })
}

// @Tags SysUser
// @Summary 设置用户信息
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.SysUser true "设置用户信息"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"修改成功"}"
// @Router /staff/setSelfInfo [put]
export const setSelfInfo = (data) => {
  return service({
    url: '/staff/setSelfInfo',
    method: 'put',
    data: data
  })
}

// @Tags User
// @Summary 设置用户权限
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body api.setUserAuthorities true "设置用户权限"
// @Success 200 {string} json "{"success":true,"data":{},"msg":"修改成功"}"
// @Router /staff/setUserAuthorities [post]
export const setUserAuthorities = (data) => {
  return service({
    url: '/staff/setUserAuthorities',
    method: 'post',
    data: data
  })
}

// @Tags User
// @Summary 获取用户信息
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Success 200 {string} json "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /staff/getUserInfo [get]
export const getUserInfo = () => {
  return service({
    url: '/staff/getUserInfo',
    method: 'get'
  })
}

export const resetPassword = (data) => {
  return service({
    url: '/staff/resetPassword',
    method: 'post',
    data: data
  })
}

File: server/router/system/sys_user.go
package system

import (
	v1 "github.com/flipped-aurora/gin-vue-admin/server/api/v1"
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type SysUserRouter struct{}

func (s *SysUserRouter) InitSysUserRouter(Router *gin.RouterGroup) {
	sysUserRouter := Router.Group("staff").Use(middleware.OperationRecord())
	sysyUserRouterWithoutRecord := Router.Group("staff")
	baseApi := v1.ApiGroupApp.SystemApiGroup.BaseApi
	{
		sysUserRouter.POST("admin_register", baseApi.Register)               // 管理员注册账号
		sysUserRouter.POST("changePassword", baseApi.ChangePassword)         // 用户修改密码
		sysUserRouter.POST("setUserAuthority", baseApi.SetUserAuthority)     // 设置用户权限
		sysUserRouter.DELETE("deleteUser", baseApi.DeleteUser)               // 删除用户
		sysUserRouter.PUT("setUserInfo", baseApi.SetUserInfo)                // 设置用户信息
		sysUserRouter.PUT("setSelfInfo", baseApi.SetSelfInfo)                // 设置自身信息
		sysUserRouter.POST("setUserAuthorities", baseApi.SetUserAuthorities) // 设置用户权限组
		sysUserRouter.POST("resetPassword", baseApi.ResetPassword)           // 设置用户权限组
	}
	{
		sysyUserRouterWithoutRecord.POST("getUserList", baseApi.GetUserList) // 分页获取用户列表
		sysyUserRouterWithoutRecord.GET("getUserInfo", baseApi.GetUserInfo)  // 获取自身信息
	}
}
